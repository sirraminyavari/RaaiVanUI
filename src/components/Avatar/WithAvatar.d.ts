import { FunctionComponent } from 'react';

export interface IWithAvatar {
  componentURLProp?: string;
  Component?: FunctionComponent<Record<string | 'src', any>>;
  AvatarSVGObject?: Record<string, string>;
}

declare const WithAvatar: (
  IWithAvatar
) => FunctionComponent<Record<string, any>>;

export default WithAvatar;
