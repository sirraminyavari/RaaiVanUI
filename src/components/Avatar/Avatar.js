/**
 * An Avatar for showing user image and access to lts profile
 */
import PropTypes from 'prop-types';
import AvatarIcon from 'components/Icons/UserIcon/User';
import * as Styled from './Avatar.styles';

/**
 * @typedef PropType
 * @type {Object}
 * @property {number} [radius] - The radius of the avatar's circule.
 * @property {string} userImage - User's image for avatar.
 */

/**
 *  @description Renders an avatar component.
 * @component
 * @param {PropType} props -Props that pass to avatar.
 */
const Avatar = (props) => {
  const { radius, userImage } = props;
  return (
    <Styled.AvatarContainer>
      {userImage ? (
        <Styled.AvatarImage radius={radius} src={userImage} alt="user-avatar" />
      ) : (
        <AvatarIcon size={radius} />
      )}
    </Styled.AvatarContainer>
  );
};

Avatar.propTypes = {
  radius: PropTypes.number,
  userImage: PropTypes.string,
  linkTo: PropTypes.string,
};

Avatar.defaultProps = {
  radius: 40,
};

export default Avatar;
