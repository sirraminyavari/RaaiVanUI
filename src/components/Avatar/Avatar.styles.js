import styled from 'styled-components';

export const AvatarContainer = styled.div`
  display: flex;
  place-items: center;
  border-radius: 50%;
  overflow: hidden;
  color: #fff;
  border: 0.2rem solid ${({ color }) => (!!color ? color : '#fff')};
`;

export const AvatarImage = styled.img`
  width: ${({ radius }) => `${radius}px`};
  height: ${({ radius }) => `${radius}px`};
  border-radius: 50%;
`;
