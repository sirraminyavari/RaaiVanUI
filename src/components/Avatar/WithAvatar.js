import * as AvatarSVGS from 'assets/images/avatars/AvatarProfileAssets';
import { avatarIconURL } from 'helpers/helpers';

/**
 * @component - A HOC component converting base64 encoded AvatarName and ImageURL properties and injecting an Avatar URL or the profile image URL as a desired prop argument to passed component
 *
 * @param {string} props.componentURLProp - The component's URL receiving property name (e.g. "src" is receiving property name for <img/> html tag)
 * @param {JSX.Element} props.Component - The component that needs an URL for displaying an Avatar or Image
 * @param {object} [props.AvatarSVGsObject] - An Object the of Avatar URLs and Names.
 * @return {JSX.Element}
 * @example
 * ```jsx
 * import * as AvatarSVGS from 'assets/images/avatars/AvatarProfileAssets';
 * import AvatarComponent from 'components/Avatar/Avatar';
 * import WithAvatar from 'components/Avatar/WithAvatar';
 *
 *const Avatar = WithAvatar({Component:AvatarComponent, componentURLProp:'userImage',AvatarSVGsObject:AvatarSVGS});
 * ...
 *
 * //* pass user object containing "ImageURL" and "AvatarName" to the "userObject" attribute
 * <Avatar  userObject={window.RVGlobal.CurrentUser} {...restProps}/>
 *
 * ```
 */

const WithAvatar =
  ({ Component, componentURLProp, AvatarSVGsObject = AvatarSVGS }) =>
  (props) => {
    const { userObject, [componentURLProp]: urlProp, ...rest } = props;

    const newProps = {
      ...rest,
      [componentURLProp]: avatarIconURL({
        userObject,
        defaultURL: urlProp,
        AvatarSVGsObject,
      }),
    };

    return (
      <>
        <Component {...newProps} />
      </>
    );
  };

WithAvatar.displayName = 'WithAvatar';

export default WithAvatar;
