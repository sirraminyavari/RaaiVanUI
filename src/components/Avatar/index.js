import { Link } from 'react-router-dom';
import { GoPerson as AvatarIcon } from 'react-icons/go';
import { AvatarWrapper, AvatarImage } from './Avatar.styles';

const Avatar = ({ radius, userImage }) => {
  return (
    <AvatarWrapper as={Link} to="/user">
      {userImage ? (
        <AvatarImage radius={radius} src={userImage} alt="user-avatar" />
      ) : (
        <AvatarIcon size={radius} color="#fff" />
      )}
    </AvatarWrapper>
  );
};

export default Avatar;
