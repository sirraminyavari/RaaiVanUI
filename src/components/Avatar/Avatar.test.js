import { render, cleanup, screen } from '@testing-library/react';
import Avatar from './Avatar';

beforeEach(() => {
  global.GlobalUtilities = {
    random_str: function () {
      return 'string';
    },
  };
});

afterEach(() => {
  delete global.GlobalUtilities;
  cleanup();
});

describe('User Avatar Component', () => {
  it('renders without crashing', () => {
    render(<Avatar />);
  });

  it('renders avatar image', () => {
    const wrapper = render(<Avatar userImage="url/image" />);

    expect(wrapper).toMatchSnapshot();

    const imgElement = screen.getByTestId('avatar-image');
    expect(imgElement).toBeInTheDocument();

    wrapper.debug();
  });

  it('renders avatar default icon', () => {
    const wrapper = render(<Avatar />);

    expect(wrapper).toMatchSnapshot();

    const iconElement = screen.getByTestId('avatar-icon');
    expect(iconElement).toBeInTheDocument();

    wrapper.debug();
  });
});
