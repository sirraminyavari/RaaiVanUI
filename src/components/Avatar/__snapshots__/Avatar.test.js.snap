// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Avatar Component renders avatar default icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ hBhCGt"
      >
        <span>
          string
        </span>
        <svg
          data-testid="avatar-icon"
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 12 16"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 14.002a.998.998 0 0 1-.998.998H1.001A1 1 0 0 1 0 13.999V13c0-2.633 4-4 4-4s.229-.409 0-1c-.841-.62-.944-1.59-1-4 .173-2.413 1.867-3 3-3s2.827.586 3 3c-.056 2.41-.159 3.38-1 4-.229.59 0 1 0 1s4 1.367 4 4v1.002z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ hBhCGt"
    >
      <span>
        string
      </span>
      <svg
        data-testid="avatar-icon"
        fill="currentColor"
        height="40"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 12 16"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 14.002a.998.998 0 0 1-.998.998H1.001A1 1 0 0 1 0 13.999V13c0-2.633 4-4 4-4s.229-.409 0-1c-.841-.62-.944-1.59-1-4 .173-2.413 1.867-3 3-3s2.827.586 3 3c-.056 2.41-.159 3.38-1 4-.229.59 0 1 0 1s4 1.367 4 4v1.002z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`User Avatar Component renders avatar image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ hBhCGt"
      >
        <span>
          string
        </span>
        <img
          alt="user-avatar"
          class="sc-gsTCUz dTjxRR"
          data-testid="avatar-image"
          radius="40"
          src="url/image"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ hBhCGt"
    >
      <span>
        string
      </span>
      <img
        alt="user-avatar"
        class="sc-gsTCUz dTjxRR"
        data-testid="avatar-image"
        radius="40"
        src="url/image"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
